components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/api/APIStore.java

```diff
@@ -211,82 +214,128 @@ SubscriptionResponse addApiSubscription(String apiId, String applicationId, Stri
+            APIMgtResourceNotFoundException;
 
     /**
      * Update a comment
      *
-     * @param comment new Comment object
+     * @param comment   new Comment object
      * @param commentId the id of the comment which needs to be updated
-     * @param apiId UUID of the api the comment belongs to
-     * @throws APIManagementException if failed to update a comment
+     * @param apiId     UUID of the api the comment belongs to
+     * @param username  username of the consumer
+     * @throws APICommentException if failed to update a comment
+     * @throws APIMgtResourceNotFoundException if api or comment not found
      */
-    void updateComment(Comment comment, String commentId, String apiId) throws APIManagementException;
+    void updateComment(Comment comment, String commentId, String apiId, String username) throws APICommentException,
+            APIMgtResourceNotFoundException;
 
     /**
      * Retrieve list of comments for a given apiId
      *
      * @param apiId UUID of the api
-     * @throws APIManagementException if failed to retrieve all comments for an api
      * @return a list of comments for the api
+     * @throws APICommentException if failed to retrieve all comments for an api
+     * @throws APIMgtResourceNotFoundException if api not found
+     */
+    List<Comment> getCommentsForApi(String apiId) throws APICommentException, APIMgtResourceNotFoundException;
+
+    /**
+     * Retrieve Individual Comment based on Comment ID
+     *
+     * @param commentId UUID od the comment
+     * @param apiId     UUID of the API
+     * @return Comment Object.
+     * @throws APICommentException if failed to retrieve comment from data layer
+     * @throws APIMgtResourceNotFoundException if api or comment was not found
+     */
+    Comment getCommentByUUID(String commentId, String apiId) throws APICommentException,
+            APIMgtResourceNotFoundException;
+
+    /**
+     * Creates a new rating
+     *
+     * @param apiId  UUID of the api
+     * @param rating the rating object
+     * @return UUID of the newly created or updated rating
+     * @throws APIRatingException if failed to add rating
+     * @throws  APIMgtResourceNotFoundException if api not found
      */
-    List<Comment> getCommentsForApi(String apiId) throws APIManagementException;
+    String addRating(String apiId, Rating rating) throws APIRatingException, APIMgtResourceNotFoundException;
 
     /**
+     * Retrieves a rating given its UUID
+     *
+     * @param apiId    UUID of the api
+     * @param ratingId UUID of the rating
+     * @return rating object
+     * @throws APIRatingException if failed to get rating
+     * @throws  APIMgtResourceNotFoundException if api or rating not found
+     */
+    Rating getRatingByUUID(String apiId, String ratingId) throws APIRatingException, APIMgtResourceNotFoundException;
+
+    /**
+     * Retrieve Average Rating based on the API ID
```

Comment:
Better if you can elaborate more on what is this average. Precision and any specialties in calculation